src/avram/queryable.cr

```diff
@@ -245,9 +245,19 @@ module Avram::Queryable(T)
     @preloads << block
   end
 
+  def cache_store
+    Fiber.current.query_cache
+  end
+
+  private def cache_key : String
+    [query.statement, query.args].join(':')
+  end
+
   def results : Array(T)
-    exec_query.tap do |records|
-      preloads.each(&.call(records))
+    cache_store.fetch(cache_key, as: Array(T)) do
```

Comment:
This doesn't log anything if it returns the cached value. We can't add logging to the block because that gets cached, so would it make sense to have `fetch` do its own logging?